server/apps/archived/archived.py

```diff
@@ -164,36 +164,44 @@ def on_delete(self, doc):
         doc[config.ID_FIELD] = id_field
 
     def delete(self, lookup):
+        if app.testing and len(lookup) == 0:
+            super().delete(lookup)
+            return
+
+    def on_deleted(self, doc):
+        pass
+
+    def update(self, id, updates, original):
```

Comment:
1. Agreed that validating item to be killed via DELETE isn't REST complaint but for just validation I want to avoid creation of one more endpoint/resource/service.

2. Until user clicks on Kill button as mentioned in the second step we shouldn't make changes to the item in archived.

3. Unlike production the items in archived doesn't change in little time as they make an entry into archived after they expire in production.